test/Microsoft.OpenApi.Readers.Tests/V3Tests/OpenApiDiscriminatorTests.cs

```diff
@@ -25,19 +27,20 @@ public async Task ParseBasicDiscriminatorShouldSucceed()
             memoryStream.Position = 0;
 
             // Act
-            var discriminator = OpenApiModelFactory.Load<OpenApiDiscriminator>(memoryStream, OpenApiSpecVersion.OpenApi3_0, OpenApiConstants.Yaml, new(), out var diagnostic, SettingsFixture.ReaderSettings);
+            var openApiDocument = new OpenApiDocument();
+            var discriminator = OpenApiModelFactory.Load<OpenApiDiscriminator>(memoryStream, OpenApiSpecVersion.OpenApi3_0, OpenApiConstants.Yaml, openApiDocument, out var diagnostic, SettingsFixture.ReaderSettings);
 
             // Assert
             Assert.Equivalent(
-                new OpenApiDiscriminator
-                {
-                    PropertyName = "pet_type",
-                    Mapping =
+               new OpenApiDiscriminator
+               {
+                   PropertyName = "pet_type",
+                   Mapping =
                     {
-                            ["puppy"] = "#/components/schemas/Dog",
-                            ["kitten"] = "Cat"
+                            ["puppy"] = new OpenApiSchemaReference("Dog", openApiDocument),
+                            ["monster"] = new OpenApiSchemaReference("schema.json" , openApiDocument, "https://gigantic-server.com/schemas/Monster/schema.json")
```

Comment:
```suggestion
                            ["monster"] = new OpenApiSchemaReference("Sully" , openApiDocument, "https://gigantic-server.com/schemas/Monster/schema.json")
```

Just for sanity, so we differentiate the schema name from the file. I suspect this is going to impact your test file.

Reply:
In this case, wouldn't the test fall under [ExternalSchemaReference Tests](https://github.com/microsoft/OpenAPI.NET/blob/main/test/Microsoft.OpenApi.Readers.Tests/V31Tests/OpenApiComponentsTests.cs) where we add this scenario `[InlineData("https://gigantic-server.com/schemas/Monster/schema.json#/components/schemas/Sully", "Sully", "https://gigantic-server.com/schemas/Monster/schema.json")]`

Reply:
let's keep the original then and ALSO add this test case, this way we're covered for:
- internal
- external single component files
- external multiple component files
what do you think?